x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/ApiKeyService.java

```diff
@@ -273,6 +275,8 @@ XContentBuilder newDocument(SecureString apiKey, String name, Authentication aut
             .field("metadata", authentication.getUser().metadata())
             .field("realm", authentication.getLookedUpBy() == null ?
                 authentication.getAuthenticatedBy().getName() : authentication.getLookedUpBy().getName())
+            .field("realm_type", authentication.getLookedUpBy() == null ?
+                authentication.getAuthenticatedBy().getType() : authentication.getLookedUpBy().getType())
```

Comment:
All this duplication bothers me. Can we just put a method like `getRealmThatReallyResolvedTheUser()` on `Authentication`?
Obviously there's some work to do on that name.

Reply:
> would you open a new issue for that, and tag it both `:Security` and `:Core/Features/Ingest` so we can get some discussion on that?
> 
Done #51799

Reply:
Add a new method [getNominalRealm](https://github.com/elastic/elasticsearch/pull/51305/files#diff-f1298d89a800df414923a8d291565e55R79) to Authentication class and replace the usages accordingly.